### ISSUE TITLE
Refresh query on variable change

### ISSUE THREAD

**Issue by slindersAmsterdam** [VolkovLabs/volkovlabs-echarts-panel#121]
When changing a variable, (all) queries used for datasource get refreshed. Also when the variable does not affect the query.
Variables based on a query do NOT get refreshed when another variable changes.
Request:
- Only refresh a query when input variables change 
   or add an option to 'query options' where you can select the correct parameters. That would be here:

![afbeelding](https://user-images.githubusercontent.com/88667072/212635320-341fce25-c711-4852-a100-5d21c945731c.png)

- Add an extra option in the pull down menu of a variable based on a query: 'Refresh on variable X change'

![afbeelding](https://user-images.githubusercontent.com/88667072/212635142-92945f4b-f6a9-4b98-b7ad-56cd4b6a529b.png)

**Comment by slindersAmsterdam** [comment]
The data source is a query on Microsoft SQL server. It gets updated every time a grafana variable changes.
A Grafana variable can be used to change the query of the datasource (and then the query gets updated automatically), but a Grafana variable can also be used to change the filter in Apache Echarts (replace variable). In the latter case the query should not be updated in Grafana (but it does).

**Comment by mikhail-vl** [comment]
@slindersAmsterdam What do you think if you can display the `select` element to choose a country directly on the Apache ECharts panel instead of using Dashboard Variables? Will it work for you?

Similar to the button in the top-left corner?
![Screen Shot 2023-01-16 at 21 19 39](https://user-images.githubusercontent.com/47795110/213503726-75523088-0ada-474d-9acb-2470fe26ad45.png)

**Comment by slindersAmsterdam** [comment]
That could be an option that could work.  Is there an example for that?

**Comment by mikhail-vl** [comment]
@slindersAmsterdam This is the code one of the community members shared to get a DOM from Apache EChart's instance and add a button for his use case. It will be a part of the upcoming blog and I was planning to add a `select` element, hopefully over the weekend.

Adding a `select` element on DOM will be similar and should be easy to google. If you make it work, please share.

```javascript
const myFunction = () => {
  alert('myFunction() called!');
}

//Get the DOM for the panel
let dom = echartsInstance.getDom();

//Get the child nodes
let nodeList = dom.childNodes;

//If we haven't added our controls to the DOM yet
if (nodeList.length < 2) {
  //Create a new button
  const btn = document.createElement("button");

  //Establish button name
  const textnode = document.createTextNode("My Button");
  btn.appendChild(textnode);

  //Add click event handler
  btn.addEventListener("click", myFunction);

  //Create a new <div>
  const div = document.createElement("div");

  //Add button to <div>
  div.appendChild(btn);

  //Insert new <div> ahead of existing chart div
  dom.insertBefore(div, dom.firstChild);
}
```

**Comment by slindersAmsterdam** [comment]
Ok I still have to learn a lot, I can't make it work (but I'm curious if you can add a select element). Still it puzzles me why the data source query gets refreshed in Grafana every time a grafana variable changes. If the variable only interacts with the echart panel, the datasource should not be refreshed. Maybe I'm missing something here..

**Comment by mikhail-vl** [comment]
@slindersAmsterdam, I will look closely later this week.

**Comment by mikhail-vl** [comment]
@slindersAmsterdam We just released Apache ECharts 4.3.0 and added an example with a select list you was looking for: https://volkovlabs.io/blog/echarts-panel-4.3.0-20230417/#panel-variables.

  Data Sources with Text inputs are always refreshing when variable changes, there is no workaround it seems. So, using a select list is a good option.
